From 5d3abedb7097375f882be7a403753aae7cb17d7b Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Horia=20Geant=C4=83?= <horia.geanta@nxp.com>
Date: Fri, 16 Dec 2016 12:25:06 +0200
Subject: [PATCH 590/666] crypto: caam - manually set
 CRYPTO_DEV_FSL_CAAM_{IMX,LE}
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

In order to accommdate the fact that multi_v7_defconfig starts
being used also for ARMv8 AArch32 (LS1012A, LS1043A etc.),
CRYPTO_DEV_FSL_CAAM_{IMX,LE} are no longer being automatically set
to "y".
Instead, CONFIG_CRYPTO_DEV_FSL_CAAM_{IMX,LE}=y must specifically
be added in defconfig files where needed (i.MX6/7, LS1021A).

*Note: patch is not upstreamable. A solution that would be accepted
upstream should detect at runtime the platform (i.MX) and
endianness.

Signed-off-by: Horia GeantÄƒ <horia.geanta@nxp.com>
[Original patch taken from SDK-V2.0-1703]
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm/configs/imx_v6_v7_defconfig | 2 ++
 drivers/crypto/caam/Kconfig          | 4 ++--
 2 files changed, 4 insertions(+), 2 deletions(-)

diff --git a/arch/arm/configs/imx_v6_v7_defconfig b/arch/arm/configs/imx_v6_v7_defconfig
index bb6fa56..8e1c3f5 100644
--- a/arch/arm/configs/imx_v6_v7_defconfig
+++ b/arch/arm/configs/imx_v6_v7_defconfig
@@ -386,6 +386,8 @@ CONFIG_PROVE_LOCKING=y
 # CONFIG_ARM_UNWIND is not set
 CONFIG_SECURITYFS=y
 CONFIG_CRYPTO_DEV_FSL_CAAM=y
+CONFIG_CRYPTO_DEV_FSL_CAAM_LE=y
+CONFIG_CRYPTO_DEV_FSL_CAAM_IMX=y
 CONFIG_CRYPTO_DEV_SAHARA=y
 CONFIG_CRC_CCITT=m
 CONFIG_CRC_T10DIF=y
diff --git a/drivers/crypto/caam/Kconfig b/drivers/crypto/caam/Kconfig
index e36aeac..acb5210 100644
--- a/drivers/crypto/caam/Kconfig
+++ b/drivers/crypto/caam/Kconfig
@@ -142,8 +142,8 @@ config CRYPTO_DEV_FSL_CAAM_RNG_API
 	  will be called caamrng.
 
 config CRYPTO_DEV_FSL_CAAM_IMX
-	def_bool SOC_IMX6 || SOC_IMX7D
-	depends on CRYPTO_DEV_FSL_CAAM
+	bool "CAAM running on an i.MX6/7 platform"
+	depends on CRYPTO_DEV_FSL_CAAM && (SOC_IMX6 || SOC_IMX7D)
 
 config CRYPTO_DEV_FSL_CAAM_DEBUG
 	bool "Enable debug output in CAAM driver"
-- 
2.7.4

